### ISSUE TITLE
how to use with sigmoid as activation function when meeting binary classification segmentation task

### ISSUE THREAD

**Issue by KeanuWongves** [LIVIAETS/boundary-loss#52]
Hi, thanks for ur great work.
But I'm confused, when I have to use it in a binary calssification segmentation task, like only foreground and background.
So , according to the readme, the dist_label should be like [batch_size,2,h,w]
but when I using sigmoid as activation function, the logits should be like [batch_size,1,h,w],
the channels dimention is not match, I dont know is there something I get wrong?
Hope to ur replys, Thanks.

**Comment by q671383789** [comment]
acctually I realized that under this circumstance I dont have to make the dist_label to be[bs,2,h,w] just use gt [bs,1,h,w] and self.idc = 0 will worked am I right? cause the logits after sigmoid is the probs for my foreground and the gt is just the probs for foreground

**Comment by KeanuWongves** [comment]
So, I dont know is it right,         
```python
pc = probs[:, self.idc, ...].type(torch.float32)
dc = dist_maps[:, self.idc, ...].type(torch.float32)
```
change to 
```python
pc = probs[:, 0, ...].type(torch.float32)
dc = dist_maps[:, self.idc, ...].type(torch.float32)
```

**Comment by KeanuWongves** [comment]
Also, I want to know, in the onehot2dis function, dtype is none so the res would be all integer, if it is ok I change it to float cause all my target is only few pixel, and I think this would be more precisely

**Comment by HKervadec** [comment]
Hi,

If you are using a sigmoid, then I am assuming that you are in a binary case. In that case the one-hot encoding of the label is redundant, but to keep changes to a minimum, for you case, you could go with:
```python
pc = probs[:, 0, ...].type(torch.float32)
dc = dist_maps[:, 1, ...].type(torch.float32)
```
as `self.idc` would always have the value of `[1]` (it is mostly useful in multi-class).

> Also, I want to know, in the onehot2dis function, dtype is none so the res would be all integer, if it is ok I change it to float cause all my target is only few pixel, and I think this would be more precisely

Yes that would work. I think I added the dtype option at the time, so that the code could handle distances maps with varying resolution across axises. 

But to be useful, you would need to pass a `resolution` argument, otherwise you would only fill a float tensor with integers.

Cheers,

Hoel
